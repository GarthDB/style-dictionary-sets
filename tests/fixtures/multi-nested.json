{
  "accent-background-color": {
    "default": {
      "sets": {
        "light": {
          "value": "{accent-color-900}"
        },
        "dark": {
          "value": "{accent-color-500}"
        }
      }
    }
  },
  "accent-color-900": {
    "sets": {
      "spectrum": {
        "value": "{blue-900}"
      },
      "express": {
        "value": "{indigo-900}"
      }
    }
  },
  "accent-color-500": {
    "sets": {
      "spectrum": {
        "value": "{blue-500}"
      },
      "express": {
        "value": "{indigo-500}"
      }
    }
  },
  "blue-900": {
    "sets": {
      "light": {
        "value": "rgb(2, 101, 220)"
      },
      "dark": {
        "value": "rgb(114, 183, 249)"
      }
    }
  },
  "indigo-900": {
    "sets": {
      "light": {
        "value": "rgb(82, 88, 228)"
      },
      "dark": {
        "value": "rgb(167, 170, 255)"
      }
    }
  },
  "blue-500": {
    "sets": {
      "light": {
        "value": "rgb(120, 187, 250)"
      },
      "dark": {
        "value": "rgb(3, 103, 224)"
      }
    }
  },
  "indigo-500": {
    "sets": {
      "light": {
        "value": "rgb(172, 175, 255)"
      },
      "dark": {
        "value": "rgb(85, 91, 231)"
      }
    }
  }
}
