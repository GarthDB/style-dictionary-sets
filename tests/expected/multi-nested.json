{
  "accent-background-color": {
    "default": {
      "sets": {
        "light": {
          "ref": "{accent-color-900}",
          "sets": {
            "spectrum": {
              "ref": "{blue-900}",
              "sets": {
                "light": {
                  "value": "rgb(2, 101, 220)"
                },
                "dark": {
                  "value": "rgb(114, 183, 249)"
                }
              }
            },
            "express": {
              "ref": "{indigo-900}",
              "sets": {
                "light": {
                  "value": "rgb(82, 88, 228)"
                },
                "dark": {
                  "value": "rgb(167, 170, 255)"
                }
              }
            }
          }
        },
        "dark": {
          "ref": "{accent-color-500}",
          "sets": {
            "spectrum": {
              "ref": "{blue-500}",
              "sets": {
                "light": {
                  "value": "rgb(120, 187, 250)"
                },
                "dark": {
                  "value": "rgb(3, 103, 224)"
                }
              }
            },
            "express": {
              "ref": "{indigo-500}",
              "sets": {
                "light": {
                  "value": "rgb(172, 175, 255)"
                },
                "dark": {
                  "value": "rgb(85, 91, 231)"
                }
              }
            }
          }
        }
      }
    },
    "hover": {
      "sets": {
        "light": {
          "ref": "{accent-color-900}",
          "sets": {
            "spectrum": {
              "ref": "{blue-900}",
              "sets": {
                "light": {
                  "value": "rgb(2, 101, 220)"
                },
                "dark": {
                  "value": "rgb(114, 183, 249)"
                }
              }
            },
            "express": {
              "ref": "{indigo-900}",
              "sets": {
                "light": {
                  "value": "rgb(82, 88, 228)"
                },
                "dark": {
                  "value": "rgb(167, 170, 255)"
                }
              }
            }
          }
        },
        "dark": {
          "ref": "{accent-color-500}",
          "sets": {
            "spectrum": {
              "ref": "{blue-500}",
              "sets": {
                "light": {
                  "value": "rgb(120, 187, 250)"
                },
                "dark": {
                  "value": "rgb(3, 103, 224)"
                }
              }
            },
            "express": {
              "ref": "{indigo-500}",
              "sets": {
                "light": {
                  "value": "rgb(172, 175, 255)"
                },
                "dark": {
                  "value": "rgb(85, 91, 231)"
                }
              }
            }
          }
        }
      }
    }
  },
  "accent-color-900": {
    "sets": {
      "spectrum": {
        "ref": "{blue-900}",
        "sets": {
          "light": {
            "value": "rgb(2, 101, 220)"
          },
          "dark": {
            "value": "rgb(114, 183, 249)"
          }
        }
      },
      "express": {
        "ref": "{indigo-900}",
        "sets": {
          "light": {
            "value": "rgb(82, 88, 228)"
          },
          "dark": {
            "value": "rgb(167, 170, 255)"
          }
        }
      }
    }
  },
  "accent-color-500": {
    "sets": {
      "spectrum": {
        "ref": "{blue-500}",
        "sets": {
          "light": {
            "value": "rgb(120, 187, 250)"
          },
          "dark": {
            "value": "rgb(3, 103, 224)"
          }
        }
      },
      "express": {
        "ref": "{indigo-500}",
        "sets": {
          "light": {
            "value": "rgb(172, 175, 255)"
          },
          "dark": {
            "value": "rgb(85, 91, 231)"
          }
        }
      }
    }
  },
  "blue-900": {
    "sets": {
      "light": {
        "value": "rgb(2, 101, 220)"
      },
      "dark": {
        "value": "rgb(114, 183, 249)"
      }
    }
  },
  "indigo-900": {
    "sets": {
      "light": {
        "value": "rgb(82, 88, 228)"
      },
      "dark": {
        "value": "rgb(167, 170, 255)"
      }
    }
  },
  "blue-500": {
    "sets": {
      "light": {
        "value": "rgb(120, 187, 250)"
      },
      "dark": {
        "value": "rgb(3, 103, 224)"
      }
    }
  },
  "indigo-500": {
    "sets": {
      "light": {
        "value": "rgb(172, 175, 255)"
      },
      "dark": {
        "value": "rgb(85, 91, 231)"
      }
    }
  }
}
